modules/databricks-loader/src/main/scala/com/snowplowanalytics/snowplow/loader/databricks/Databricks.scala

```diff
@@ -88,10 +103,16 @@ object Databricks {
                 sql"""COPY INTO $frTableName
                       FROM (SELECT _C0::VARCHAR(512) RUN_ID FROM '$frPath')
                       FILEFORMAT = CSV""";
-              case Statement.EventsCopy(path, _, _) =>
-                val frTableName = Fragment.const(EventsTable.withSchema(config.storage.schema))
-                val frPath      = Fragment.const0(s"$path/output=good")
-                sql"""COPY INTO $frTableName FROM '$frPath' FILEFORMAT = PARQUET COPY_OPTIONS('MERGESCHEMA' = 'TRUE')""";
+              case Statement.EventsCopy(path, _, columns) =>
+                val frTableName     = Fragment.const(EventsTable.withSchema(config.storage.schema))
+                val frPath          = Fragment.const0(s"$path/output=good")
+                val frSelectColumns = Fragment.const0(columns.mkString(",") + ", current_timestamp() as load_tstamp")
+                sql"""COPY INTO $frTableName
+                      FROM (
+                        SELECT $frSelectColumns from '$frPath'
+                      )
+                      FILEFORMAT = PARQUET
+                      COPY_OPTIONS('MERGESCHEMA' = 'TRUE')""";
```

Comment:
Have you checked this still works in the following situations:

- New batch has a column which is not yet present in the table
- Existing table contains a column which is missing from the new batch of parquet

Reply:
Yep, both of these cases are working okay.